scout/adapter/mongo/case.py

```diff
@@ -197,86 +194,149 @@ def _set_diagnosis_query(self, query: Dict[str, Any], query_term: str):
-            if query_term != "":
-                query["phenotype_groups.phenotype_id"] = query_term
-            else:
-                query["$or"] = [
-                    {"phenotype_groups": {"$size": 0}},
-                    {"phenotype_groups": {"$exists": False}},
-                ]
-        if query_field == "cohort":
-            query["cohorts"] = query_term
+        def set_similar_pheno_query(query: dict, query_value: str):
+            """Updates the query with a certain value."""
+            self._set_similar_phenotype_query(
+                query, "similar_pheno", query_value, owner or collaborator
+            )
 
-        if query_term != "" and query_field in ["similar_case", "similar_pheno"]:
-            self._set_similar_phenotype_query(query, query_field, query_term, owner or collaborator)
+        def set_pinned_gene_query(query: dict, query_value: str):
+            """Updates the query with a certain value."""
+            self._set_genes_of_interest_query(
+                query=query, query_field="pinned", query_value=query_value
+            )
 
-        if query_term != "" and query_field in ["pinned", "causative"]:
-            self._set_genes_of_interest_query(query, query_field, query_term)
+        def set_causative_gene_query(query: dict, query_value: str):
+            """Updates the query with a certain value."""
+            self._set_genes_of_interest_query(
+                query=query, query_field="causative", query_value=query_value
+            )
 
-        if query_field == "user":
-            query_terms = query_term.split(" ")
+        def set_user_query(query: dict, query_value: str):
+            # Handling the 'user' field query separately
+            query_terms = query_value.split(" ")
             user_query = {
                 "$and": [{"name": {"$regex": term, "$options": "i"}} for term in query_terms]
             }
             users = self.user_collection.find(user_query)
             query["assignees"] = {"$in": [user["_id"] for user in users]}
 
+        if isinstance(name_query, str):
+            # Example: status -> Comes from a GET request from dashboard cases
+            query_field, query_value = name_query.split(":", 1)
+            set_case_item_query(
+                query=query, query_field=query_field.strip(), query_value=query_value.strip()
+            )
+        else:
+            # POST request form from more advanced case search from cases page
+            for query_field in [
+                "case",
+                "exact_pheno",
+                "exact_dia",
+                "synopsis",
+                "panel",
+                "status",
+                "tags",
+                "track",
+                "pheno_group",
+                "cohort",
+                "pinned",
+                "causative",
+                "user",
+                "similar_case",  # In order to be able to sort results by phenotype similarity, keep this at the bottom
```

Comment:
I have moved these 2 at the bottom so whatever happens before setting the query _id values, the order by similarity will be maintained (hopefully)